File: armstrong-number/armstrong-number.py
# Armstrong Number Checker in Python

num = 371  # Change this to test other numbers
num_str = str(num)
num_len = len(num_str)

sum_of_powers = sum(int(digit) ** num_len for digit in num_str)

if sum_of_powers == num:
    print(f"{num} is an Armstrong number.")
else:
    print(f"{num} is NOT an Armstrong number.")

File: armstrong-number/armstrong-number.js
// Armstrong Number Checker in JavaScript

let num = 153; // Change this to test other numbers
let numStr = num.toString();
let numLen = numStr.length;

let sum = 0;
for (let ch of numStr) {
    sum += Math.pow(parseInt(ch), numLen);
}

if (sum === num) {
    console.log(num + " is an Armstrong number.");
} else {
    console.log(num + " is NOT an Armstrong number.");
}
